### PR TITLE
Vet clinic database: query multiple tables

### PR DESCRIPTION
**Pull request by BezzB**
Project requirements
Create a table named owners with the following columns:
id: integer (set it as autoincremented PRIMARY KEY)
full_name: string
age: integer
Create a table named species with the following columns:
id: integer (set it as autoincremented PRIMARY KEY)
name: string
Modify animals table:
Make sure that id is set as autoincremented PRIMARY KEY
Remove column species
Add column species_id which is a foreign key referencing species table
Add column owner_id which is a foreign key referencing the owners table
Remember all this goes in schema.sql file.
Insert the following data into the owners table:
Sam Smith 34 years old.
Jennifer Orwell 19 years old.
Bob 45 years old.
Melody Pond 77 years old.
Dean Winchester 14 years old.
Jodie Whittaker 38 years old.
Insert the following data into the species table:
Pokemon
Digimon
Modify your inserted animals so it includes the species_id value:
If the name ends in "mon" it will be Digimon
All other animals are Pokemon
Modify your inserted animals to include owner information (owner_id):
Sam Smith owns Agumon.
Jennifer Orwell owns Gabumon and Pikachu.
Bob owns Devimon and Plantmon.
Melody Pond owns Charmander, Squirtle, and Blossom.
Dean Winchester owns Angemon and Boarmon.
Remember these insertions and modifications happen in data.sql file.
Write queries (using JOIN) to answer the following questions:
What animals belong to Melody Pond?
List of all animals that are pokemon (their type is Pokemon).
List all owners and their animals, remember to include those that don't own any animal.
How many animals are there per species?
List all Digimon owned by Jennifer Orwell.
List all animals owned by Dean Winchester that haven't tried to escape.
Who owns the most animals?
Remember all these should be written in queries.sql file.
Screenshot from 2023-09-20 17-52-33
Screenshot from 2023-09-20 18-28-48
Screenshot from 2023-09-20 18-29-32
Screenshot from 2023-09-20 18-31-34
Screenshot from 2023-09-20 18-32-29
Screenshot from 2023-09-20 18-33-26
Screenshot from 2023-09-20 18-34-48
Screenshot from 2023-09-20 18-35-52

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -66,4 +66,99 @@ UPDATE animals
...
+VALUES ('Melody Pond', 77);
+
+-- Insert Dean Winchester
+INSERT INTO owners (full_name, age)
+VALUES ('Dean Winchester', 14);
+
+-- Insert Jodie Whittaker
+INSERT INTO owners (full_name, age)
+VALUES ('Jodie Whittaker', 38);
+
```
Comment (Bantealem) [inline queries.sql]: Kindly that only the query that helps to analyze the data should be put in the `queries.sql` file. Currently, these are used to populate the table and should be put in the `data.sql` file according to the project requirement. Kindly, fix it.

image

File: queries.sql [hunk 2]
```diff
@@ -66,4 +66,99 @@ UPDATE animals
...
+CREATE TABLE species (
+    id serial PRIMARY KEY,
+    name varchar(255)
+);
+
+CREATE TABLE owners (
+    id serial PRIMARY KEY,
+    full_name varchar(255),
+    age integer
+);
```
Comment (Bantealem) [inline queries.sql]: Everything that changes the table structure should be put in the `schema .sql` file based on the project requremnt. Please  move these code to the `schema.sql` file.
image

**Review by Bantealem — CHANGES_REQUESTED** [review]
# Status: Change Required
Hi @BezzB, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- [x] Professional Reame
- [x] Descriptive PR summary and title.
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [ ] It would be better to make the branch name descriptive full since it will be more formal and clear for other developer.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Owusu-Desmond — APPROVED** [review]
Hi @BezzB , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
📌 Good commit messages ✅
📌 Professional README file ✅

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**